models/public/cocosnet/model.yml

```diff
@@ -0,0 +1,170 @@
+    source: https://raw.githubusercontent.com/vacancy/Synchronized-BatchNorm-PyTorch/5768ead395820b8a625cbe0da0bb9f949748a5dd/sync_batchnorm/batchnorm.py
+  - name: models/networks/sync_batchnorm/replicate.py
+    size: 3226
+    sha256: 4498b873404fce05202d4f891262e249a95733816d3877e3aee3d2401138a6af
+    source: https://raw.githubusercontent.com/vacancy/Synchronized-BatchNorm-PyTorch/5768ead395820b8a625cbe0da0bb9f949748a5dd/sync_batchnorm/replicate.py
+  - name: models/networks/sync_batchnorm/comm.py
+    size: 4449
+    sha256: 9d2e776a0398e5ac29ce13d43b7475300f70f409e56f3c38428029000557a57a
+    source: https://raw.githubusercontent.com/vacancy/Synchronized-BatchNorm-PyTorch/5768ead395820b8a625cbe0da0bb9f949748a5dd/sync_batchnorm/comm.py
+  - name: options/__init__.py
+    size: 72
+    sha256: 12b307709a025b0afdfdd9e7ad584fbf21d9170be7c0a3f8432e24abae471af7
+    source: https://raw.githubusercontent.com/microsoft/CoCosNet/33f98d092407094a15a08b0555d6f5359490cd3e/options/__init__.py
+  - name: options/base_options.py
+    size: 11951
+    sha256: a7516bb137c52857edc5f9dea240bc79eabe7e774820b26f0fef7f793870d475
+    source: https://raw.githubusercontent.com/microsoft/CoCosNet/33f98d092407094a15a08b0555d6f5359490cd3e/options/base_options.py
+  - name: options/test_options.py
+    size: 1217
+    sha256: 0dad4f1a8541d1316df94fdc41c06295e1601e91613eeeb29416181b7d72205c
+    source: https://raw.githubusercontent.com/microsoft/CoCosNet/33f98d092407094a15a08b0555d6f5359490cd3e/options/test_options.py
+  - name: ckpt/latest_net_Corr.pth
+    size: 237624725
+    sha256: 5deef28927dfc56e18c8510ffcdf0280b558987a38a2b455cda62242ecbe7971
+    source:
+      $type: google_drive
+      id: 1yNCapBY7YqMoKeyVWwTwceqrEKHpUzYN
+  - name: ckpt/latest_net_G.pth
+    size: 387069636
+    sha256: 9a77b52d0c34a279d6b4e7eff4bd00077543f7ffc2806b5b2a32222cfbccc655
+    source:
+      $type: google_drive
+      id: 1zVG4SnR1uI8bC8fagZrgxNYCI5GRmJuR
+postprocessing:
+  - $type: regex_replace
+    file: models/networks/normalization.py
+    pattern: 'x.var\(dim=1, keepdim=True\).add\(epsilon\).sqrt\(\)'
+    replacement: 'x.std(dim=1, keepdim=True)'
+  - $type: regex_replace
+    file: models/networks/correspondence.py
+    pattern: '/ self.opt.down\)'
+    replacement: '// self.opt.down)'
+  - $type: regex_replace
+    file: options/base_options.py
+    pattern: 'import data'
+    replacement: ''
+  - $type: regex_replace
+    file: options/base_options.py
+    pattern: '# modify dataset-related parser options'
+    replacement: "'''"
+  - $type: regex_replace
+    file: options/base_options.py
+    pattern: 'parser = dataset_option_setter\(parser, self.isTrain\)'
+    replacement: "'''"
+  - $type: regex_replace
+    file: options/base_options.py
+    pattern: '# set gpu ids'
+    replacement: 'opt.gpu_ids = "-1"'
+  - $type: regex_replace
+    file: options/base_options.py
+    pattern: 'opt = parser.parse_args'
+    replacement: 'opt, _ = parser.parse_known_args'
```

Comment:
All this option-related hackery seems pretty hairy. Maybe it would be easier, instead of using `TestOptions`, to create a `types.SimpleNamespace` with all the options and pass that to the model factory instead?